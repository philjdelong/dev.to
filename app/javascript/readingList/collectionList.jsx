import { h, Component } from 'preact';
import PropTypes from 'prop-types';
import { Collection } from './components/Collection';

export class CollectionList extends Component {
  constructor({ collections }) {
    super({ collections });

    this.state = {
      collections,
    };
  }

  render() {
    const { collections } = this.state;
    const collectionsToRender = collections.map(collection => {
      return (
        <Collection
          key={collection.id}
          name={collection.name}
          tags={collection.tag_list}
          slug={collection.slug}
        />
      );
    });

    return (
      <section
        className="collection-cont results results--loaded"
        id="reading-collection"
      >
        <div className="results-header collection-header">
          {collections.length
            ? `Collections (${collections.length})`
            : 'Collections'}
          <a className="new-collection" href="/readingcollections/new">
            +
          </a>
        </div>
        {collectionsToRender}
      </section>
    );
  }
}

CollectionList.propTypes = {
  collections: PropTypes.arrayOf(PropTypes.object).isRequired,
};
