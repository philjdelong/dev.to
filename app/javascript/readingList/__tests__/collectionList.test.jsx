import { h } from 'preact';
import render from 'preact-render-to-json';
import { CollectionList } from '../collectionList';

describe('<ContentList />', () => {
  it('renders properly', () => {
    const mockCollections = [
      { id: 1, name: 'first_test_post' },
      { id: 2, name: 'second_test_post' },
    ];
    const tree = render(<CollectionList collections={mockCollections} />);
    expect(tree).toMatchSnapshot();
  });
});
