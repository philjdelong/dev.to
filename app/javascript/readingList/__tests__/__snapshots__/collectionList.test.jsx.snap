// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentList /> renders properly 1`] = `
<section
  class="collection-cont results results--loaded"
>
  <div
    class="results-header"
  >
    Collections (2)
  </div>
  <section
    class="collection item-wrapper"
  >
    <div
      class="item"
    >
      <p
        class="item-title"
      >
        first_test_post
      </p>
    </div>
  </section>
  <section
    class="collection item-wrapper"
  >
    <div
      class="item"
    >
      <p
        class="item-title"
      >
        second_test_post
      </p>
    </div>
  </section>
</section>
`;
